assets/js/modules/optimize/index.js

```diff
@@ -20,44 +20,36 @@
  * WordPress dependencies
  */
 import { addFilter } from '@wordpress/hooks';
-import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import OptimizeSetup from '../../modules/optimize/setup';
-import { fillFilterWithComponent, getModulesData } from '../../util';
+import { fillFilterWithComponent } from '../../util';
 
-const slug = 'optimize';
-const modulesData = getModulesData();
-if ( modulesData.optimize.active ) {
-	/**
-	 * Add components to the settings page.
-	 */
-	addFilter( `googlesitekit.ModuleSettingsDetails-${ slug }`,
-		'googlesitekit.OptimizeModuleSettingsDetails',
-		fillFilterWithComponent( OptimizeSetup, {
-			onSettingsPage: true,
-		} ) );
-
-	/**
-	 * Add component to the setup wizard
-	 */
-	addFilter( `googlesitekit.ModuleSetup-${ slug }`,
-		'googlesitekit.OptimizeModuleSetupWizard',
-		fillFilterWithComponent( OptimizeSetup, {
-			onSettingsPage: false,
-		} ) );
+/**
+ * Internal dependencies
+ */
+import './datastore';
 
-	/**
-	 * Add data to the congrats setup Win Notification for display.
-	 */
-	addFilter( `googlesitekit.SetupWinNotification-${ slug }`,
-		'googlesitekit.OptimizeSetupWinNotification', ( winData ) => {
-			winData.description = __( 'To set up experiments and see the results, go to ', 'google-site-kit' );
-			winData.learnMore.label = 'Optimize';
-			winData.learnMore.url = 'https://optimize.withgoogle.com/';
-			return winData;
-		} );
-}
+/**
+ * Add components to the settings page.
+ */
+addFilter(
+	'googlesitekit.ModuleSettingsDetails-optimize',
+	'googlesitekit.OptimizeModuleSettingsDetails',
+	fillFilterWithComponent( OptimizeSetup, {
+		onSettingsPage: true,
+	} )
+);
 
+/**
+ * Add component to the setup wizard
+ */
+addFilter(
+	'googlesitekit.ModuleSetup-optimize',
+	'googlesitekit.OptimizeModuleSetupWizard',
+	fillFilterWithComponent( OptimizeSetup, {
+		onSettingsPage: false,
+	} )
+);
```

Comment:
This code should remain in `index.legacy.js` since it is still using a legacy component (see #1621 for changing that). This file should only have the datastore for now. (Also note that this still needs to be wrapped into the `modulesData.optimize.active` check like before.)

Reply:
`index.js` and `index.legacy.js` refactored as requested